snappy/kernel_os.go

```diff
@@ -222,13 +222,14 @@ func SyncBoot() error {
 		return fmt.Errorf("failed to run SyncBoot: %s", err)
 	}
 
+	overlord := &Overlord{}
```

Comment:
hah! I was wondering the same, but the other way around: why not just use an overlord there. But I'm fine with landing this as is and waiting for @mvo5's answer/patch later :-)